hartshorn-core/src/main/java/org/dockbox/hartshorn/core/domain/Exceptional.java

```diff
@@ -578,17 +607,18 @@ public Throwable unsafeError() {
      *
      * @return The type of the value, or {@code null}
      */
+    @Nullable
     public Class<?> type() {
         return this.present() ? this.value.getClass() : null;
     }
-
+    
     /**
      * Checks if the value is present, and if it is, checks if it is equal to the given value.
      *
      * @param other The value to check for equality
      * @return {@code true} if the value is present and equal to {@code other}, otherwise {@code false}
      */
-    public boolean equal(final Object other) {
+    public boolean equal(@NonNull final Object other) {
```

Comment:
`other` may be null

Reply:
If `other` is nullable, then you would expect it to return true if `value` is null and `other` is null right? However, due to the `this.present()` check, it would instead return false:
![image](https://user-images.githubusercontent.com/38820160/147300675-c58c7b7e-f03e-4469-ad1b-1c42b03dbe71.png)
Would you like me to replace it with `return Objects.equal(this.orNull(), other);`

Reply:
Comparing `null` can yield unexpected behavior, though when it matters it's safe to assume a condition can include `this.present()` manually (so it becomes `typeA.present() && typeA.equal(typeB)`). Your suggestion sounds good 👍🏼